src/Vue/Resources/assets/src/register_controller.ts

```diff
@@ -9,18 +9,21 @@
 
 'use strict';
 
-export function registerVueControllerComponents(context: __WebpackModuleApi.RequireContext) {
+export function registerVueControllerComponents(contexts: any) {
     const vueControllers: { [key: string]: object } = {};
 
     const importAllVueComponents = (r: __WebpackModuleApi.RequireContext) => {
         r.keys().forEach((key) => (vueControllers[key] = r(key).default));
     };
 
-    importAllVueComponents(context);
+    [].concat(contexts).forEach((context: __WebpackModuleApi.RequireContext) => importAllVueComponents(context));
 
     // Expose a global Vue loader to allow rendering from the Stimulus controller
     (window as any).resolveVueComponent = (name: string): object => {
-        const component = vueControllers[`./${name}.vue`];
+        const component = Object.values(
+            Object.fromEntries(Object.entries(vueControllers).filter(([key]) => key.endsWith(`${name}.vue`)))
+        )[0];
```

Comment:
tbh, this makes my head spin a little bit.

If I import 2 contexts, and each contains a `Hello.vue`, the 2nd one will "win" over the first, correct?

With the new code, is the "key" inside `vueControllers` now a longer path (e.g. `common/Hello.vue` instead of just `Hello.vue`)? What makes these code changes needed?